doc/sphinx/proof-engine/vernacular-commands.rst

```diff
@@ -1308,10 +1168,28 @@ Inlining hints for the fast reduction machines
 Registering primitive operations
 ````````````````````````````````
 
-.. cmd:: Primitive @ident__1 := #@ident__2.
+.. cmd:: Primitive @ident {? : @term } := # @ident__prim
```

Comment:
Would there be a way to remove the space after `#`? What about having the grammar checker ignore the missing spaces just like it ignores the additional subscripts?

Reply:
Sure.  Actually it's a tweak to how productions are printed, similar to those for "." and "?".

Reply:
OK cool, then could you also include `@` in this category?

Reply:
Does the lexer require no space between `@` and what follows?  And if not, why suppress the space?

Also, at the moment the space is suppressed only before `ident` tokens.  While that can be changed, let's be restrained in adding grammar-specific things in the code.

`"@"` appears only in these productions for various nonterminals.

```
| "@" qualid OPT univ_annot LIST0 term1
| "@" qualid OPT univ_annot
| term0 ".(" "@" qualid LIST0 ( term1 ) ")"
| "@" "?" ident LIST1 ident
| "@" qualid LIST0 pattern1
```

Reply:
OK never mind. Spaces are allowed after the @.